### PR TITLE
Add hyperVGeneration enum property to ImageProperties

### PR DESCRIPTION
**Pull request by vamuzumd**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by NelsonDaniel** [comment]
@vamuzumd, **I left some review comments.**

Also, if you haven't taken a look at our new onboarding experience at OpenAPIHub, it's a convenient way to create your PR when you're copying from an existing API version or when you're editing your existing specs. If you have any feedback or questions, feel free to use the feedback button on top of the site for help. Thanks!"

**Comment by hyonholee** [comment]
Can you merge this PR?  We need a fixed SDK ASAP.  Thanks.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json [hunk 1]
```diff
@@ -6800,6 +6812,12 @@
           "readOnly": true,
           "type": "string",
           "description": "The provisioning state."
+        },
+        "hyperVGeneration": {
+          "hyperVGenerationType": {
```
[the remark this answers is not in the document]
Comment (vamuzumd) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: > @vamuzumd (1) Is your intention to add hyperVGenerationType as a property of hyperVGeneration? If that's the case it should be:
> 
> ```json
>  "hyperVGeneration":  {
>     "properties": {
>         "hyperVGenerationType": {
>         ...
>          }
>    }
> }
> ```
> 
> or
> (2) does hyperVGeneration match the definition of HyperVGenerationType? In that case eliminate 6817 and 6819.

HyperVGenerationType is an enum and HyperVGeneration is a wrapper that encapsulates it, we cannot merge them or include HyperVGenerationType as a property.

[the remark this answers is not in the document]
Comment (NelsonDaniel) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: @vamuzumd that is not valid swagger. It should be:
``` json
"hyperVGeneration" : {
    "$ref": "#/definitions/HyperVGenerationType",
    "description": "Gets the HyperVGenerationType of the image"
}
```

**Review by NelsonDaniel — CHANGES_REQUESTED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json (hunk 1 above)
Comment (NelsonDaniel) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: @vamuzumd that is not valid swagger. It should be:
``` json
"hyperVGeneration" : {
    "$ref": "#/definitions/HyperVGenerationType",
    "description": "Gets the HyperVGenerationType of the image"
}
```

**Review by NelsonDaniel — CHANGES_REQUESTED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json (hunk 1 above)
Comment (vamuzumd) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: yes, I will make the change

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json [hunk 2]
```diff
@@ -6800,6 +6812,10 @@
           "readOnly": true,
           "type": "string",
           "description": "The provisioning state."
+        },
+        "hyperVGeneration": {
+          "$ref": "#/definitions/HyperVGenerationType"
```
Comment (jhendrixMSFT) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: Missing trailing ','

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Add missing comma to fix syntax errors

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5942 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5942/commits/418d1cd3f802dab5158150688f309fcf9b1eed1d

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by hyonholee** [comment]
Thanks, @jhendrixMSFT.

@NelsonDaniel , could you merge this PR?  Thanks.

**Pull request merged**